test/extended/networking/commatrix.go

```diff
@@ -0,0 +1,252 @@
+		docURLStatusIsNotFound := resp.StatusCode == http.StatusNotFound
+		o.Expect(docURLStatusIsNotFound).To(o.BeFalse())
+
+		g.By("preparing for commatrices generation")
+		epExporter, err := endpointslices.New(cs)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		utilsHelpers := utils.New(cs)
+
+		g.By("generating new commatrix")
+		newComMatrixCreator, err := commatrixcreator.New(epExporter, "", "", env, deployment)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		newComMatrix, err := newComMatrixCreator.CreateEndpointMatrix()
+		o.Expect(err).ToNot(o.HaveOccurred())
+		newComMatrix.WriteMatrixToFileByType(utilsHelpers, "new-commatrix", types.FormatCSV, deployment, artifactsDir)
+
+		g.By("get documented commatrix")
+		fp := filepath.Join(artifactsDir, "doc-commatrix.csv")
+		createCSVFromUrl(docCommatrixVersionedUrl, fp)
+		docComMatrixCreator, err := commatrixcreator.New(epExporter, fp, types.FormatCSV, env, deployment)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		docComDetailsList, err := docComMatrixCreator.GetComDetailsListFromFile()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("Filter documented commatrix for diff generation")
+		// if cluster is a SNO exclude standard deployment type's static entries in diff generation
+		if isSNO {
+			// Exclude static entries of standard deployment type and all worker nodes entries.
+			docComDetailsList = excludeGivenStaticEntries(docComDetailsList, &types.ComMatrix{Matrix: types.StandardStaticEntries}, &types.ComMatrix{Matrix: docComDetailsList})
+		} else {
+			// Exclude specific master entries that are documented as both worker and master,
+			// and are open only on the worker node in standard clusters.
+			docComDetailsList = excludeGivenStaticEntries(docComDetailsList, &types.ComMatrix{Matrix: StandardExcludedMasterComDetails}, nil)
+
+		}
+
+		// if cluster is running on BM exclude Cloud static entries in diff generation
+		// else cluster is running on Cloud and exclude BM static entries in diff generation.
+		if isBM {
+			docComDetailsList = excludeGivenStaticEntries(docComDetailsList,
+				&types.ComMatrix{Matrix: types.CloudStaticEntriesMaster},
+				&types.ComMatrix{Matrix: types.CloudStaticEntriesWorker})
+		} else {
+			docComDetailsList = excludeGivenStaticEntries(docComDetailsList,
+				&types.ComMatrix{Matrix: types.BaremetalStaticEntriesMaster},
+				&types.ComMatrix{Matrix: types.BaremetalStaticEntriesWorker})
+		}
+		docComMatrix := &types.ComMatrix{Matrix: docComDetailsList}
+
+		g.By("generating diff between matrices for testing purposes")
+		diff := matrixdiff.Generate(newComMatrix, docComMatrix)
+		diffStr, err := diff.String()
+		o.Expect(err).ToNot(o.HaveOccurred())
+		err = os.WriteFile(filepath.Join(artifactsDir, "doc-diff-new"), []byte(diffFileComments+diffStr), 0644)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("comparing new and documented commatrices")
+		areEqual := comMatricesAreEqual(*newComMatrix, *docComMatrix)
```

Comment:
We should use the diff object here, if something is missing in the diff package let's add it